includes/data-stores/class-wcs-customer-store-cached-cpt.php

```diff
@@ -114,7 +114,9 @@ public function get_users_subscription_ids( $user_id ) {
 		}
 
 		// Sort results in order to keep consistency between cached results and queried results.
-		rsort( $subscription_ids );
+		if ( ! empty( $subscription_ids ) ) {
```

Comment:
```suggestion
		if ( is_array( $subscription_ids ) ) {
```

Thoughts on changing this to is_array()?

Or maybe return an empty array if it's not in the expected array format:

```
if ( ! is_array( $subscription_ids ) ) {
    return [];
}

// Sort results in order to keep consistency between cached results and queried results.
rsort( $subscription_ids );

return $subscription_ids;
```

Benefits to this is we're also not going to try call rsort on strings and other non-array types.
We're also validating the output of this function but returning an empty list if `$subscription_ids` is something other than array

Reply:
Thanks @mattallan. I am going for the 2nd suggestion and returning an empty array. Addressed in [dd5c692](https://github.com/Automattic/woocommerce-subscriptions-core/pull/607/commits/dd5c6929cd368febc3b026a1465a937bbc11d9f9)